File: app/modules/ytdlp/android/src/test/java/expo/modules/ytdlp/UrlTest.kt
package expo.modules.ytdlp

import org.junit.Assert.assertEquals
import org.junit.Test

class UrlTest {
    @Test
    fun `normalizeUrl prepends https to a schemeless link`() {
        assertEquals("https://youtube.com/watch?v=jNQXAC9IVRw", normalizeUrl("youtube.com/watch?v=jNQXAC9IVRw"))
    }

    @Test
    fun `normalizeUrl leaves an already-schemed link unchanged`() {
        assertEquals("http://youtube.com/watch?v=x", normalizeUrl("http://youtube.com/watch?v=x"))
    }

    @Test
    fun `normalizeUrl trims surrounding whitespace before checking for a scheme`() {
        assertEquals("https://youtu.be/x", normalizeUrl("  youtu.be/x  "))
    }

    @Test
    fun `normalizeUrl prepends https to a schemeless www link`() {
        assertEquals("https://www.youtube.com/watch?v=x", normalizeUrl("www.youtube.com/watch?v=x"))
    }
}

File: server/__tests__/url.test.ts
import { describe, expect, it } from "vitest";
import { normalizeUrl } from "../src/url.js";

describe("normalizeUrl", () => {
  it("prepends https:// to a schemeless link", () => {
    expect(normalizeUrl("youtube.com/watch?v=jNQXAC9IVRw")).toBe("https://youtube.com/watch?v=jNQXAC9IVRw");
  });

  it("leaves an already-schemed link unchanged", () => {
    expect(normalizeUrl("http://youtube.com/watch?v=x")).toBe("http://youtube.com/watch?v=x");
  });

  it("trims surrounding whitespace before checking for a scheme", () => {
    expect(normalizeUrl("  youtu.be/x  ")).toBe("https://youtu.be/x");
  });

  it("prepends https:// to a schemeless www. link", () => {
    expect(normalizeUrl("www.youtube.com/watch?v=x")).toBe("https://www.youtube.com/watch?v=x");
  });
});
